### PR TITLE
Update docker images for manylinux

### PR DESCRIPTION
**Pull request by veblush**
### General Linux docker images
- Updated `auditwheel` to the latest version, `2.1.1`.

### Manylinux docker images
- Added new `manylinux2010_x86` docker image.
- Added cython for python3.8.
- Removed `auditwheel` installation because it's already there.

### Python artifact build script
- Updated the script to use  `AUDITWHEEL_PLAT` baked in manylinux docker images instead of a regular expression to have a more precise validation.

### CODE REVIEW

**Comment by veblush** [comment]
Tests are triggered:
- [grpc_build_artifacts_multiplatform](https://sponge.corp.google.com/invocation?id=d9cab816-8e67-4f4e-b07e-e3a498e092fa&searchFor)
- [grpc_build_artifacts_extra_release](https://sponge.corp.google.com/invocation?id=bdb22944-e32b-4d9a-968e-87fb9a877134&searchFor=)

File: tools/dockerfile/grpc_artifact_python_manylinux1_x64/Dockerfile [hunk 1]
```diff
@@ -28,11 +28,4 @@ RUN /opt/python/cp34-cp34m/bin/pip install cython
...
 RUN /opt/python/cp37-cp37m/bin/pip install cython
-
-####################################################
-# Install auditwheel with fix for namespace packages
-RUN git clone https://github.com/pypa/auditwheel /usr/local/src/auditwheel
-RUN cd /usr/local/src/auditwheel && git checkout 2.1
-RUN /opt/python/cp36-cp36m/bin/pip install /usr/local/src/auditwheel
-RUN rm /usr/local/bin/auditwheel
-RUN cd /usr/local/bin && ln -s /opt/python/cp36-cp36m/bin/auditwheel
+RUN /opt/python/cp38-cp38/bin/pip install cython
```
Comment (lidizheng) [inline tools/dockerfile/grpc_artifact_python_manylinux1_x64/Dockerfile]: Good to know the "m" suffix is no longer needed: https://bugs.python.org/issue36707

File: tools/run_tests/artifacts/build_artifact_python.sh [hunk 2]
```diff
@@ -79,12 +79,12 @@ ${SETARCH_CMD} "${PYTHON}" tools/distrib/python/grpcio_tools/setup.py bdist_whee
 if [ "$GRPC_BUILD_MANYLINUX_WHEEL" != "" ]
 then
   for wheel in dist/*.whl; do
-    "${AUDITWHEEL}" show "$wheel" | tee /dev/stderr |  grep -E -w 'manylinux(1|2010)_(x86_64|i686)'
+    "${AUDITWHEEL}" show "$wheel" | tee /dev/stderr |  grep -E -w "$AUDITWHEEL_PLAT"
```
Comment (lidizheng) [inline tools/run_tests/artifacts/build_artifact_python.sh]: Can you explain about how is this env is set? I can only find its usage.
https://github.com/pypa/auditwheel/blob/c6ff475c3325ae393ffb5872b73cb4583d914a06/auditwheel/main_repair.py

Comment (veblush) [reply tools/run_tests/artifacts/build_artifact_python.sh]: Here is a definition of this env: https://github.com/pypa/manylinux/blob/99c19386e670f2aecbe4a5d0397a10a04e2d1eeb/docker/Dockerfile-x86_64#L6

**Pull request merged**